src/Core/Extensible.php

```diff
@@ -553,8 +557,21 @@ public function getExtensionInstances()
 
             if ($extensions) {
                 foreach ($extensions as $extension) {
-                    $instance = Injector::inst()->get($extension);
-                    $this->extension_instances[get_class($instance)] = $instance;
+                    $name = $extension;
+                    // Allow service names of the form "%$ServiceName"
+                    if (substr($name, 0, 2) == '%$') {
+                        $name = substr($name, 2);
+                    }
+
+                    if (strstr($name, '(')) {
+                        list($name) = ClassInfo::parse_class_spec($name);
+                    }
+                    $name = trim($name);
+                    try {
```

Comment:
An exception thrown would have a larger performance cost than a `class_exists` check. We should avoid using exceptions in non-error conditions.

Reply:
done